### PR TITLE
Adding missing filelists, adding more tighter photon ID criteria for photons other than the loose ID.

### PR DESCRIPTION
**Pull request by alpana-hep**
This request contains the filelists for MC samples which were missing. These lists are stored under the following directory - "Production/test/condorSub/FilesToReprocess'. This directory also contains the file lists for the MC samples which needs to be reprocessed after adding mindR(photon, q/g) variable.

This request is adding additional two branches corresponding to different photon IDs in the final ntuple.

### CODE REVIEW

**Comment by kpedro88** [comment]
immediate comments:
1. do not duplicate existing condor dictionaries.
2. do not put new condor dictionaries in a separate folder; put them directly in condorSub.
3. do not specify unnecessary arguments in the new condor dictionaries (remove `jetsconstituents=2 boostedsemivisible=True tchannel=True"`, which are needed for semivisible jet analyses, not SUSY photon analyses).
4. fix misspelling "Summaer" -> "Summer" in new dictionary filenames.
5. it looks like you did not `git add` the new sample file lists in `TreeMaker/Production/python/*/`.

File: Production/test/runMakeTreeFromMiniAOD_cfg.py [hunk 1]
```diff
@@ -37,6 +37,17 @@
 if len(theMaker.localera)>0:
 	eralist.append(getattr(TMeras,theMaker.localera))
 process = cms.Process("RA2EventSelection",*eralist)
+#from RecoEgamma.EgammaTools.EgammaPostRecoTools import setupEgammaPostRecoSeq
```
Comment (kpedro88) [inline Production/test/runMakeTreeFromMiniAOD_cfg.py]: delete commented-out code

File: Production/test/runMakeTreeFromMiniAOD_cfg.py [hunk 2]
```diff
@@ -37,6 +37,17 @@
...
+from TreeMaker.Utils.EgammaPostRecoTools import setupEgammaPostRecoSeq
```
Comment (kpedro88) [inline Production/test/runMakeTreeFromMiniAOD_cfg.py]: you need to add the lines to obtain `EgammaPostRecoTools` to `TreeMaker/setup.sh` (as previously noted via email)

File: Production/test/runMakeTreeFromMiniAOD_cfg.py [hunk 3]
```diff
@@ -37,6 +37,17 @@
 if len(theMaker.localera)>0:
 	eralist.append(getattr(TMeras,theMaker.localera))
 process = cms.Process("RA2EventSelection",*eralist)
+#from RecoEgamma.EgammaTools.EgammaPostRecoTools import setupEgammaPostRecoSeq
+from TreeMaker.Utils.EgammaPostRecoTools import setupEgammaPostRecoSeq 
+setupEgammaPostRecoSeq(process,era='2018-UL',
```
Comment (kpedro88) [inline Production/test/runMakeTreeFromMiniAOD_cfg.py]: should `era='2018-UL'` be used for every data-taking year in UL? or are there different eras for each year?

File: Production/test/runMakeTreeFromMiniAOD_cfg.py [hunk 4]
```diff
@@ -37,6 +37,17 @@
...
+                       runVID=True,
+                       eleIDModules=['RecoEgamma.ElectronIdentification.Identification.cutBasedElectronID_Fall17_94X_V2_cff',
```
Comment (kpedro88) [inline Production/test/runMakeTreeFromMiniAOD_cfg.py]: is it correct that `Fall17` configurations are used for the `'2018-UL'` era?

File: Production/test/runMakeTreeFromMiniAOD_cfg.py (hunk 2 above)
Comment (kpedro88) [reply Production/test/runMakeTreeFromMiniAOD_cfg.py]: also, please move all of this code from `runMakeTreeFromMiniAOD_cfgpy` to `TreeMaker/TreeMaker/python/doPhotons.py`

File: TreeMaker/python/doPhotons.py [hunk 5]
```diff
@@ -24,6 +24,8 @@ def doPhotonVars(self,process):
     self.VectorBool.append("goodPhotons:fullID(Photons_fullID)")
     self.VectorBool.append("goodPhotons:electronFakes(Photons_electronFakes)")
     self.VarsBool.append("goodPhotons:hasGenPromptPhoton(hasGenPromptPhoton)")
+    self.VectorDouble.append("goodPhotons:mvaValuesIDFall17V2(Photons_mvaValuesIDFall17V2)")
```
Comment (kpedro88) [inline TreeMaker/python/doPhotons.py]: suggest just calling these branches `Photons_mvaValuesID` and `Photons_cutBasedID`

File: Utils/src/PhotonIDisoProducer.cc [hunk 6]
```diff
@@ -231,7 +235,15 @@ PhotonIDisoProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::Event
 
     std::vector<float> vCov = clusterTools_.localCovariances( *(iPhoton.superCluster()->seed()) ); 
     const float sieie = (isnan(vCov[0]) ? 0. : sqrt(vCov[0])); 
-    
+    double phoMVAID = iPhoton.userFloat("PhotonMVAEstimatorRunIIFall17v2Values");
+    int tmpphoIDbit = -1;
+    bool isPassLoose  = iPhoton.photonID("cutBasedPhotonID-Fall17-94X-V2-loose");
```
Comment (kpedro88) [inline Utils/src/PhotonIDisoProducer.cc]: these strings should be part of the python configuration

**Comment by alpana-hep** [comment]
Hi Kevin,
I have fixed most of your comments on the PR. Following are not yet resolved and need your help on these.

>should era='2018-UL' be used for every data-taking year in UL? or are there different eras for each year?
No, it should be different for different years as mentioned in this github page[1] but I am not able to figure out how this 'era' variable can be added and be used while runing the machinery. 
> is it correct that Fall17 configurations are used for the '2018-UL' era?
Yes, it should be used for all three years (twiki -[2])

> In Utils/src/PhotonIDisoProducer.cc: 
>     std::vector<float> vCov = clusterTools_.localCovariances( *(iPhoton.superCluster()->seed()) ); 
>    const float sieie = (isnan(vCov[0]) ? 0. : sqrt(vCov[0]));     
>   double phoMVAID = iPhoton.userFloat("PhotonMVAEstimatorRunIIFall17v2Values");
>    int tmpphoIDbit = -1;
>    bool isPassLoose  = iPhoton.photonID("cutBasedPhotonID-Fall17-94X-V2-loose");

>these strings should be part of the python configuration
Did you mean to replace this peice of the code by the following code? 
"in Utils/src/PhotonIDisoProducer.cc: 
 double phoMVAID = iPhoton.userFloat(mva_basedID);
 int tmpphoIDbit = -1;
bool isPassLoose  = iPhoton.photonID(cut_basedID);  "

where as the mva_basedID & cut_basedID will be defined as "PhotonMVAEstimatorRunIIFall17v2Values", "cutBasedPhotonID-Fall17-94X-V2-loose" respectivelty in  Utils/python/photonIDisoProducer_cff.py file 
[1] https://github.com/cms-egamma/EgammaPostRecoTools/blob/master/python/EgammaPostRecoTools.py#L88
[2] https://twiki.cern.ch/twiki/bin/view/CMS/CutBasedPhotonIdentificationRun2#Working_points_for_94X_and_later

**Comment by kpedro88** [comment]
TreeMaker has a specific era for each year. One example of how to use them: https://github.com/TreeMaker/TreeMaker/blob/3596f67a06876d0b279c7a8df95c38b0c14c8140/TreeMaker/python/makeTreeFromMiniAOD_cff.py#L1054-L1056

For the ID strings in the configuration, yes, that is what I meant.

**Comment by alpana-hep** [comment]
Hi
I tried to add this code such that I can change era according to the sample used. But it throws me an error saying- "Era is unknow parameter, not a global variable". I am making these changes in "TreeMaker/python/doPhotons.py".
Do you have any suggestion?
"  setupEgammaPostRecoSeq(process,era="2018-UL",
                           runVID=True,
                           eleIDModules=['RecoEgamma.ElectronIdentification.Identification.cutBasedElectronID_Fall17_94X_V2_cff',
                                        'RecoEgamma.ElectronIdentification.Identification.heepElectronID_HEEPV70_cff',
                                         'RecoEgamma.ElectronIdentification.Identification.mvaElectronID_Fall17_iso_V2_cff',
                                         'RecoEgamma.ElectronIdentification.Identification.mvaElectronID_Fall17_noIso_V2_cff'],
                           phoIDModules=['RecoEgamma.PhotonIdentification.Identification.mvaPhotonID_Fall17_94X_V2_cff',
                                         'RecoEgamma.PhotonIdentification.Identification.cutBasedPhotonID_Fall17_94X_V2_cff']
    )
    TMeras.TMUL2016.toModify(setupEgammaPostRecoSeq, era= "2016-UL")
"

**Comment by alpana-hep** [comment]
Hi Kevin,
I have made all the changes suggested by you and have pushed those changes in my local branch. Let me know if you have some other suggestions as well.

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 7]
```diff
@@ -1,5 +1,4 @@
 # $Id: makeTreeFromPAT_cff.py,v 1.16 2013/01/24 15:42:53 mschrode Exp $
-#
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py]: revert unnecessary change

File: TreeMaker/python/maker.py [hunk 8]
```diff
@@ -104,7 +104,6 @@ def __init__(self,parameters):
         self.getParamDefault("pufile",self.scenario.pufile)
         self.getParamDefault("era",self.scenario.era)
         self.getParamDefault("localera",self.scenario.localera)
-
```
Comment (kpedro88) [inline TreeMaker/python/maker.py]: revert unnecessary change

File: setup.sh [hunk 9]
```diff
@@ -3,7 +3,7 @@
 CMSSWVER=CMSSW_10_6_29_patch1
 FORK=TreeMaker
 BRANCH=Run2_UL
-ACCESS=ssh
+ACCESS=https
```
Comment (kpedro88) [inline setup.sh]: revert unnecessary change

File: setup.sh [hunk 10]
```diff
@@ -162,7 +162,7 @@ fi
 git clone ${ACCESS_GITHUB}TreeMaker/JetToolbox.git JMEAnalysis/JetToolbox -b jetToolbox_102X
 git clone ${ACCESS_GITHUB}kpedro88/CondorProduction.git Condor/Production
 git clone ${ACCESS_GITHUB}${FORK}/TreeMaker.git -b ${BRANCH}
-
+wget https://raw.githubusercontent.com/cms-egamma/EgammaPostRecoTools/master/python/EgammaPostRecoTools.py -P TreeMaker/Utils/python/
```
Comment (kpedro88) [inline setup.sh]: pin to a certain commit instead of using master

**Comment by kpedro88** [comment]
In addition, still need to fix typo "Summaer" in the ttgjets dictionary filenames.

**Comment by alpana-hep** [comment]
Hi Kevin,
I have made the changes and push it to the branch.

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 7 above)
Comment (kpedro88) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py]: still not reverted

File: setup.sh [hunk 11]
```diff
@@ -162,8 +162,8 @@ fi
 git clone ${ACCESS_GITHUB}TreeMaker/JetToolbox.git JMEAnalysis/JetToolbox -b jetToolbox_102X
 git clone ${ACCESS_GITHUB}kpedro88/CondorProduction.git Condor/Production
 git clone ${ACCESS_GITHUB}${FORK}/TreeMaker.git -b ${BRANCH}
-
-# compile
+wget https://raw.githubusercontent.com/cms-egamma/EgammaPostRecoTools/f38439658b0784e07c3c64927185919fa6594904/python/EgammaPostRecoTools.py
+ -P TreeMaker/Utils/python/
```
Comment (kpedro88) [inline setup.sh]: needs to be on the previous line

Comment (alpana-hep) [reply setup.sh]: Added it

**Comment by alpana-hep** [comment]
Hi Kevin,
Let me know if issues in PR are not yet resolved.

**Comment by sbein** [comment]
Hey @alpana-hep, @kpedro88, since the preparations for this analysis also found a strong physics motivation in the pMSSM for lower photon pT, it would be very good if we could lower the reco threshold from 30 to 20 GeV:

https://github.com/alpana-hep/TreeMaker_alp/blob/b3044a8c335abfa1c973bc7495769e331030fe1d/Utils/python/photonidisoproducer_cfi.py#L13

File: setup.sh (hunk 11 above)
Comment (kpedro88) [reply setup.sh]: that was not the comment. the line `-P TreeMaker/Utils/python/` is an argument to `wget`, so it needs to be on the same line as `wget`.

**Comment by kpedro88** [comment]
@sbein that's fine with me

**Comment by kpedro88** [comment]
@alpana-hep your fork is breaking the CI because you changed the repository name. Please change it back to just `TreeMaker`, not `TreeMaker_alp`.

**Comment by alpana-hep** [comment]
Hi Kevin, the name has been changed but I still this message -  "some of the checks were not successful". Is it the old one or I still have something incorrectly?

File: TreeMaker/python/doPhotons.py [hunk 12]
```diff
@@ -6,7 +6,20 @@ def doPhotonVars(self,process):
     process.goodPhotons = PhotonIDisoProducer.clone(
         conversionCollection   = cms.untracked.InputTag("reducedEgamma","reducedConversions",self.tagname)
     )
-
+    from TreeMaker.Utils.EgammaPostRecoTools import setupEgammaPostRecoSeq 
+    era = cms.string("2018-UL")
+    TMeras.TMUL2016.toModify(setupEgammaPostRecoSeq, era=cms.string("2016-UL"))
```
Comment (kpedro88) [inline TreeMaker/python/doPhotons.py]: https://github.com/cms-egamma/EgammaPostRecoTools/blob/f38439658b0784e07c3c64927185919fa6594904/python/EgammaPostRecoTools.py#L94-L111
this shows there are separate eras `"2016preVFP-UL"` (corresponds to `TMUL2016APV`) and `"2016postVFP-UL"` (corresponds to `TMUL2016`)

File: TreeMaker/python/doPhotons.py [hunk 13]
```diff
@@ -6,7 +6,20 @@ def doPhotonVars(self,process):
     process.goodPhotons = PhotonIDisoProducer.clone(
         conversionCollection   = cms.untracked.InputTag("reducedEgamma","reducedConversions",self.tagname)
     )
-
+    from TreeMaker.Utils.EgammaPostRecoTools import setupEgammaPostRecoSeq 
+    era = cms.string("2018-UL")
```
Comment (kpedro88) [inline TreeMaker/python/doPhotons.py]: for this to actually work, you need to use the exact syntax from here:
https://github.com/TreeMaker/TreeMaker/blob/Run2_2017_V17/TreeMaker/python/makeTreeFromMiniAOD_cff.py#L475-L478

File: TreeMaker/python/doPhotons.py (hunk 12 above)
Comment (alpana-hep) [reply TreeMaker/python/doPhotons.py]: Yes, 2016-Legacy would work for 2016-UL. I am adding it.

Comment (kpedro88) [reply TreeMaker/python/doPhotons.py]: 2016-Legacy is very old. The separate UL eras should be used as specified in my above comment.

Comment (alpana-hep) [reply TreeMaker/python/doPhotons.py]: Oh, sorry I missed it. understood now.

File: TreeMaker/python/doPhotons.py (hunk 13 above)
Comment (alpana-hep) [reply TreeMaker/python/doPhotons.py]: Added it.

File: TreeMaker/python/doPhotons.py [hunk 14]
```diff
@@ -6,7 +6,21 @@ def doPhotonVars(self,process):
...
         conversionCollection   = cms.untracked.InputTag("reducedEgamma","reducedConversions",self.tagname)
     )
-
+    from TreeMaker.Utils.EgammaPostRecoTools import setupEgammaPostRecoSeq 
+    phoPostRecoEra = cms.PSet(value = cms.string(""))
+    TMeras.TMUL2018.toModify(phoPostRecoEra, value = "2018-UL")
+    TMeras.TMUL2017.toModify(phoPostRecoEra, value = "2017-UL")
+    TMeras.TMUL2016.toModify(phoPostRecoEra, value = "2016postVFP-UL")
+    TMeras.TMUL2016APV.toModify(phoPostRecoEra,value ="2016preVFP-UL")
+    setupEgammaPostRecoSeq(process,era=phoPostRecoEra.value.value()),
```
Comment (kpedro88) [inline TreeMaker/python/doPhotons.py]: there is an extra `)` at the end of this line

**Comment by kpedro88** [comment]
The test workflow fails with the following error:
```
edm::FileInPath unable to find file EgammaAnalysis/ElectronTools/data/ScalesSmearings/Run2016_UltraLegacy_postVFP_RunFineEtaR9Gain_v3_scales.dat anywhere in the search path.
```

Indeed, I do not see that file in https://github.com/cms-data/EgammaAnalysis-ElectronTools/tree/master/ScalesSmearings (there's no `v3` in the name of the `Run2016_UltraLegacy_postVFP` file), nor is there any info about where to obtain the file on the twiki https://twiki.cern.ch/twiki/bin/view/CMS/EgammaPostRecoRecipes.

Can you follow up with the POG about where to get these files?

**Comment by kpedro88** [comment]
Alternatively, if we don't actually need these files for photon ID (I think they are for energy calibration), maybe we can disable the use of them? Maybe `runEnergyCorrections=False` when calling `setupEgammaPostRecoSeq()`?

**Comment by kpedro88** [comment]
I ran a local test and `runEnergyCorrections=False` seems to work. Please confirm that the info you need is still properly stored.

**Comment by alpana-hep** [comment]
Yes, I will follow up with the POG about these files missing.
And about the checks , I am working on it

**Comment by alpana-hep** [comment]
Yes, the fix is working for me

**Comment by alpana-hep** [comment]
Hi Kevin,
Is there something else remaining to do so that we can merge this branch?
